cli/pull.ml

```diff
@@ -125,12 +158,18 @@ let run (`Yes yes) (`No_cache no_cache) (`Repo repo) () =
       let duniverse_dir = Fpath.(repo // Config.vendor_dir) in
       Bos.OS.Dir.create duniverse_dir >>= fun _created ->
       mark_duniverse_content_as_vendored ~duniverse_dir >>= fun () ->
-      get_cache ~no_cache >>= fun cache -> pull_source_dependencies ~duniverse_dir ~cache duniverse
+      get_cache ~no_cache >>= fun cache ->
+      pull_source_dependencies ~trim_clone:(not no_sm) ~duniverse_dir ~cache duniverse >>= fun () ->
+      if no_sm then Ok () else set_git_submodules ~repo ~duniverse_dir duniverse
 
 let no_cache =
   let doc = "Run without using the duniverse global cache" in
   Common.Arg.named (fun x -> `No_cache x) Cmdliner.Arg.(value & flag & info ~doc [ "no-cache" ])
 
+let no_submodules =
+  let doc = "Run without adding the source as submodules" in
+  Common.Arg.named (fun x -> `No_submodules x) Cmdliner.Arg.(value & flag & info ~doc [ "no-submodules" ])
```

Comment:
Cmdliner already adds a `--no-` for flags, this adds `--no-submodules` and `--no-no-submodules`.

Reply:
flag removed